api-specifications.md

```diff
@@ -96,6 +104,162 @@ Create a course (anyone logged in). Used for outside Exchange.
 * Login required
 * Course already exists
 
+### /api/instructor: Course instructor management
+
+#### POST /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Add an instructor to the course. (instructors only)*
+
+##### Request (HTTP POST data)
+```
+first_name=/*instructor name*/&
+last_name=/*instructor last name*/&
+email=/*instructor email*/
+```
+
+##### Response
+```javascript
+{
+    "success": true
+}
+```
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+
+#### GET /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Gets information about a course instructor. (instructors+students)*
+
+##### Response
+```javascript
+{
+    "success": true,
+    "first_name": /*instructor name*/,
+    "last_name": /*instructor last name*/,
+    "email": /*instructor email*/
+}
+```
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+* Instructor not found
+* Removing self
+
+#### DELETE /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Remove a course instructor (instructors only)*
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+* Instructor not found
+
+### /api/instructors: List course instructors
+
+#### GET /api/instructors/&lt;course_id&gt;
+*Gets information about all course instructors. (instructors+students)*
+
+##### Response
+```javascript
+{
+    "success": true,
+    "instructors":
+    [
+        {
+            "first_name": /* instructor name*/,
+            "last_name": /* instructor lastname */,
+            "email": /* instructor email */,
+            "instructor_id": /* instructor ID */
+        },
+        ...
+    ]
+}
+```
+
+#### Error messages
+* Login required
+* Permission denied
+* Course not found
+
+### /api/student: Student management
+
+#### POST /api/student/&lt;course_id&gt;/&lt;student_id&gt;
+*Create or update a student. (instructors only)*
+
+##### Request (HTTP POST data)
+```
+first_name=/*student name*/&
+last_name=/*student last name*/&
+email=/*student email*/
+```
+
+##### Response
+```javascript
+{
+    "success": true
+}
+```
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+
+#### GET /api/student/&lt;course_id&gt;/&lt;student_id&gt;
+*Gets information about a student. (instructors+student with same student_id)*
+
+##### Response
+```javascript
+{
+    "success": true,
+    "first_name": /* student name*/,
+    "last_name": /* student lastname */,
+    "email": /* student email */
+}
+```
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+* Student not found
+
+#### DELETE /api/student/&lt;course_id&gt;/&lt;student_id&gt;
+*Removes a student (instructors only)*
```

Comment:
We need to define what happens to the submissions the student have. There are 3 possibilities:
1. Submissions are still there, visible to instructors
2. Submissions are kept in database, but invisible to anyone (when the student is added back, nothing is lost)
3. Submissions are removed from database

Reply:
I think only the student should be affected. All submissions should be kept and remain visible. Removing submissions should be left to DELETE submission, if we decide to implement that.

Reply:
LGTM (so we are using choice 1). Make sure to update API does.

Reply:
The error message will be "Removing last instructor from course"